### PR TITLE
fix the login issue at search [chm-387]

### PR DESCRIPTION
**Pull request by tahmeedshah**
### General description
fix login issue at search.

### Designs
_Link to the related design prototypes (if applicable)_

### Testing instructions
_Provide minimal instructions on how to test this PR._

* Apart from the added feature / bug fix, check overall performance, styling...

## Checklist before merging
* [ ]  Branch name / PR includes the related Jira ticket Id.
* [ ]  Tests to check core implementation / bug fix added.
* [ ]  All checks in Continuous Integration workflow pass.
* [ ]  Feature functionally tested by reviewer(s).
* [ ]  Code reviewed by reviewer(s).
* [ ]  Documentation updated (README, CHANGELOG...) (if required)

### CODE REVIEW

File: app/views/search/directives/result-view-options.js [hunk 1]
```diff
@@ -50,19 +50,18 @@ app.directive('resultViewOptions', ['$location', 'ngDialog', 'locale', '$rootSco
...
+                    if (!$rootScope.user || !$rootScope.user.isAuthenticated) {
+                        var signIn = $scope.$on('signIn', function (evt, data) {
+                            signIn();
+                        });
+                        $('#loginDialog').modal("show");
+                        return false;
+                    } 
+                    else {
+                        return true;
+                    }
```
Comment (blaisefonseca) [inline app/views/search/directives/result-view-options.js:64]: return value will not have any effect

File: app/views/search/directives/result-view-options.js [hunk 2]
```diff
@@ -50,19 +50,18 @@ app.directive('resultViewOptions', ['$location', 'ngDialog', 'locale', '$rootSco
...
-                            });
-                            $('#loginDialog').modal("show");
-                            return false;
-                        } 
-                        else {
-                            return true;
-                        }
-                    }));   
+                }
+                $scope.authenticateUser = function () { //ToDo: we can move to common
```
Comment (blaisefonseca) [inline app/views/search/directives/result-view-options.js:54]: look at app.js  `const authPlugin = createAuth ({ `, it has login function, if we are able to open the login dialog from that fn then there is no need to raise the authenticateuser event, we could show the login dialog from vue itself.

File: app/views/forms/view/record-loader.directive.js [hunk 3]
```diff
@@ -100,19 +100,18 @@ const sleep = (ms)=>new Promise((resolve)=>setTimeout(resolve, ms));
...
-							else {
-								return true;
-							}
-						}));   
+					}
+					$scope.authenticateUser = function () {
+                        if (!$rootScope.user || !$rootScope.user.isAuthenticated) {
+							var signIn = $scope.$on('signIn', function (evt, data) {
+								signIn();
+							});
```
Comment (blaisefonseca) [inline app/views/forms/view/record-loader.directive.js:108]: this is useless

File: app/views/search/directives/result-view-options.js [hunk 4]
```diff
@@ -50,19 +50,18 @@ app.directive('resultViewOptions', ['$location', 'ngDialog', 'locale', '$rootSco
...
+                    if (!$rootScope.user || !$rootScope.user.isAuthenticated) {
+                        var signIn = $scope.$on('signIn', function (evt, data) {
+                            signIn();
+                        });
```
Comment (blaisefonseca) [inline app/views/search/directives/result-view-options.js:58]: has no meaning, remove

File: app/views/search/directives/result-view-options.js (hunk 2 above)
Comment (tahmeedshah) [reply app/views/search/directives/result-view-options.js:54]: The app.js login redirects the user to the login page and changes the route. Here, we need to open the login dialog and, after logging in, stay on the same page

**Review by blaisefonseca — APPROVED** [review]

**Pull request merged**